Octane-related 3.13 blog post revisions (#282)
ember-learn/ember-blog: source/2019-09-20-ember-3-13-released.md

tags: Releases, 2019, 3, 3.13, 3.x
responsive: true
---

Today, the Ember project is releasing version 3.13 of Ember.js, Ember Data, and Ember CLI.

Ember 3.13 is the final release of Ember before Octane. Ember 3.13 is also the Octane preview. The preview allows adventurous community members to help us put the final round of polish on the edition before Octane becomes official in 3.14.

Additionally, version 3.12 of Ember is now promoted to LTS (Long Term Support). An LTS version of Ember continues to receive security updates for 9 release cycles (54 weeks) and bugfixes for 6 cycles (36 weeks). LTS releases typically occur every four minor versions. The previous LTS version of Ember was 3.8.

## The Octane Preview

The Octane features in 3.13 are stable public APIs with our normal SemVer guarantees. If you build an app using APIs in the Octane preview, it will continue to work in future versions.

So why is this a preview and not the final Octane release?

While these APIs are stable, not all Octane features are compatible with popular addons yet. We are also still updating documentation, and important tools like the Ember Inspector do not yet work with some Octane features.

> In Ember 3.14, Octane will be the recommended way to use Ember and new apps will have Octane's optional features enabled by default. The guides and tutorials will show Octane examples, and codemods will be available to help users migrate to Octane.

Reviewer: ```suggestion
In Ember 3.14, Octane will be the recommended way to use Ember and new apps will have Octane's optional features enabled by default. The [guides](https://octane-guides-preview.emberjs.com/release/) and [tutorial](https://octane-guides-preview.emberjs.com/release/tutorial/00-part-1/) will show Octane examples, and codemods will be available to help users migrate to Octane.
```

Reviewer: It may be a good idea to pull this out higher up and explicitly call out the Octane Guides Preview, and in particular, the Tutorial is a great way for existing Ember users to get up to speed on Octane. I think the tutorial is worth calling out because I wrote it and am biased 😉, but also the other sections of the guides are in various state of WIP as we get ready for the 3.14 release, but I'm pretty confident that the tutorial is up-to-date and ready for use (and feedback, of course).

**Tracked Properties and Tracked Property Updates (1 of 5)**

Tracked properties introduce a simpler and more ergonomic system for tracking state change in Ember applications. Tracked properties allow Ember to reduce its API surface area while producing code that is both more intuitive and less error-prone.

Have a look at the [guides] to learn more about tracked properties. For design details, check out [RFC 410](https://github.com/emberjs/rfcs/blob/master/text/0410-tracked-properties.md) and [RFC 478](https://github.com/emberjs/rfcs/blob/master/text/0478-tracked-properties-updates.md).

**Component Templates Co-location: Part One (2 of 5)**

Component co-location is available in Ember 3.13 when the Octane preview is enabled. This feature allows you to group a component's template and JavaScript files together in the file system. For example, a component named `user-profile` would be made up of the following files:

```
app
└── components
    ├── user-profile.hbs
    └── user-profile.js
```

> In 3.13, this feature is not available unless you enable the Octane preview. However, we expect to enable it for all apps once a [performance issue](https://github.com/ember-cli/ember-cli-htmlbars/issues/290) is addressed.

Reviewer: ```suggestion
In 3.13, this feature is not available unless you enable the Octane preview. However, we expect to enable it for all apps in the next release once a [performance issue](https://github.com/ember-cli/ember-cli-htmlbars/issues/290) is addressed.
```

No new features in Ember CLI 3.13.

---

For more details on the changes in Ember CLI 3.13 and detailed upgrade
instructions, please review the [Ember CLI  3.13.0 release page](https://github.com/ember-cli/ember-cli/releases/tag/v3.13.0).

## Ember 3.12 is an LTS Release

If you are upgrading from the previous LTS version (3.8) to 3.12 then you are getting access to these features:

* `router` service 
* Support for native classes
* The `array` helper
* Element Modifier Manager (and modifiers in general)
* Nested angle bracket component invocation
* Angle bracket invocation for built-in components
* Support for native decorators
* The `{{fn}}` and `{{on}}` modifiers

## 3.14 Beta

This release kicks off the 3.14 beta cycle for all sub-projects. We encourage our community (especially addon authors) to help test these beta builds and report any bugs before they are published as a final release in six weeks' time. The [ember-try](https://github.com/ember-cli/ember-try) addon is a great way to continuously test your projects against the latest Ember releases.

> 

Reviewer: ```suggestion

Among other things, the [`@model` named argument](https://github.com/emberjs/rfcs/blob/master/text/0523-model-argument-for-route-templates.md) will be available to route templates. While not officially an "Octane feature", this feature will available for the first time in the 3.14 release, which coincides with the Octane release. Since the Octane guides and documentation are targeting the same version, you will see them using this feature, but please not that the feature is not available in the 3.13 Octane preview.

This beta release also included a new [debug render tree API](https://github.com/emberjs/ember.js/pull/18372), which will be integrated into the Ember Inspector sometime during the beta cycle to give it full Octane support.
```

No new features in Ember CLI 3.13.

---

For more details on the changes in Ember CLI 3.13 and detailed upgrade
instructions, please review the [Ember CLI  3.13.0 release page](https://github.com/ember-cli/ember-cli/releases/tag/v3.13.0).

## Ember 3.12 is an LTS Release

If you are upgrading from the previous LTS version (3.8) to 3.12 then you are getting access to these features:

* `router` service 
* Support for native classes
* The `array` helper
* Element Modifier Manager (and modifiers in general)
* Nested angle bracket component invocation
* Angle bracket invocation for built-in components
* Support for native decorators
* The `{{fn}}` and `{{on}}` modifiers

## 3.14 Beta

This release kicks off the 3.14 beta cycle for all sub-projects. We encourage our community (especially addon authors) to help test these beta builds and report any bugs before they are published as a final release in six weeks' time. The [ember-try](https://github.com/ember-cli/ember-try) addon is a great way to continuously test your projects against the latest Ember releases.

> Among other things, the [`@model` named argument](https://github.com/emberjs/rfcs/blob/master/text/0523-model-argument-for-route-templates.md) will be available to route templates. While not officially an "Octane feature", this feature will available for the first time in the 3.14 release, which coincides with the Octane release. Since the Octane guides and documentation are targeting the same version, you will see them using this feature, but please not that the feature is not available in the 3.13 Octane preview.

Reviewer 2: please not that ➡️ please **note** that

Reviewer: Oops. Can you add it as a suggestion on #260?
